discord/ext/commands/core.py

```diff
@@ -1157,10 +1217,10 @@ def remove_command(self, name):
             # with an already existing command. If this is the case, we want to
             # make sure the pre-existing command is not removed.
             if cmd not in (None, command):
-                self.all_commands[alias] = cmd
+                self.all_commands[alias] = cmd  # type: ignore
         return command
```

Comment:
Pyright doesn't understand this guard, to remove the type ignore we'd need to change it.

Reply:
Does it understand `if cmd is not None and cmd != command`?